src/rbyte/io/video/ffmpeg_source.py

```diff
@@ -0,0 +1,41 @@
+from collections.abc import Iterable
+from pathlib import Path
+from typing import cast, override
+
+import numpy.typing as npt
+import torch
+from jaxtyping import UInt8
+from pydantic import FilePath, NonNegativeInt, validate_call
+from torch import Tensor
+from video_reader import (
+    PyVideoReader,  # pyright: ignore[reportAttributeAccessIssue, reportUnknownVariableType]
+)
+
+from rbyte.io.base import TensorSource
+
+
+class FfmpegFrameSource(TensorSource):
```

Comment:
Whats the rationale for appending `Source` ?

Reply:
not 100% sold on the naming myself but the rationale is that i updated the base protocol (`TensorSource`) to a simpler (and hopefully more pythonic) version:  
```
@runtime_checkable
class TensorSource(Protocol):
    def __getitem__(self, indexes: Iterable[Any]) -> Tensor: ...
    def __len__(self) -> int: ...
```